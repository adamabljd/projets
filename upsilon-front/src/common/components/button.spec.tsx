import React from 'react';
import { fireEvent, render, RenderResult, screen } from '@testing-library/react';
import Button, { ButtonProps } from './button.component';


const handleOnClickMock = jest.fn();

const mockButtonProps: ButtonProps = {
  onClick: handleOnClickMock,
  label: 'Test',
  variant: 'primary'
};

function renderButton(props: ButtonProps): RenderResult {
  return render(<Button {...props} />);
}

describe('Button', () => {
  describe('static rendering', () => {
    it('should render Button', () => {
      const { baseElement } = renderButton(mockButtonProps);
      expect(baseElement).toMatchSnapshot();
    });
  });
  describe('on click event', () => {
    it('should trigger on click callback function',() => {
      render(<Button {...mockButtonProps} />);
      const button = screen.getByRole('button');
      fireEvent.click(button);
      expect(handleOnClickMock).toHaveBeenCalled();
    });
  });
});
