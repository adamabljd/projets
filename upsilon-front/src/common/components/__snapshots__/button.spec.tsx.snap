// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button static rendering should render Button 1`] = `
<body>
  <div>
    <button
      class="button-component primary"
      type="button"
    >
      Test
    </button>
  </div>
</body>
`;
